src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py

```diff
@@ -193,42 +189,52 @@ def builtin_neighbors(
     offset_dim = offset_literal.value
     assert isinstance(offset_dim, str)
     iterator = transformer.visit(data)
-    table: NeighborTableOffsetProvider = transformer.offset_provider[offset_dim]
-    assert isinstance(table, NeighborTableOffsetProvider)
-
-    offset = transformer.offset_provider[offset_dim]
-    if isinstance(offset, Dimension):
+    assert isinstance(iterator, IteratorExpr)
+    field_desc = iterator.field.desc(transformer.context.body)
+
+    field_index = "__field_idx"
+    offset_provider = transformer.offset_provider[offset_dim]
+    if isinstance(offset_provider, NeighborTableOffsetProvider):
+        neighbor_check = f"{field_index} >= 0"
+    elif isinstance(offset_provider, StridedNeighborOffsetProvider):
+        neighbor_check = f"{field_index} < {field_desc.shape[offset_provider.neighbor_axis.value]}"
+    else:
+        assert isinstance(offset_provider, Dimension)
         raise NotImplementedError(
             "Neighbor reductions for cartesian grids not implemented in DaCe backend."
         )
 
+    assert transformer.context.reduce_identity is not None
+
     sdfg: dace.SDFG = transformer.context.body
     state: dace.SDFGState = transformer.context.state
 
-    shifted_dim = table.origin_axis.value
+    shifted_dim = offset_provider.origin_axis.value
 
     result_name = unique_var_name()
-    sdfg.add_array(result_name, dtype=iterator.dtype, shape=(table.max_neighbors,), transient=True)
+    sdfg.add_array(
+        result_name, dtype=iterator.dtype, shape=(offset_provider.max_neighbors,), transient=True
+    )
     result_access = state.add_access(result_name)
 
     table_name = connectivity_identifier(offset_dim)
 
     # generate unique map index name to avoid conflict with other maps inside same state
-    index_name = unique_name("__neigh_idx")
+    neighbor_index = unique_name("neighbor_idx")
     me, mx = state.add_map(
         f"{offset_dim}_neighbors_map",
-        ndrange={index_name: f"0:{table.max_neighbors}"},
+        ndrange={neighbor_index: f"0:{offset_provider.max_neighbors}"},
     )
     shift_tasklet = state.add_tasklet(
         "shift",
-        code=f"__result = __table[__idx, {index_name}]",
+        code=f"__result = __table[__idx, {neighbor_index}]",
```

Comment:
Good idea, I will try it.

Reply:
DaCe has a "TaskletFusion" transformation, which is actually quite good.
Just of curiosity have you tried to generate two maps and then let the DaCe MapFussion Transformation do the work?

However, I think what you tell me sounds more like some functionality is missing, i.e. something like `add_mapped_tasklet()` which accepts a tasklet (or a list of tasklet) that was made from external one.

Reply:
I am using `MapFusion` in `_visit_reduce`, because there we are using maps (more than 1) to build potentially multiple arrays for the input arguments to reduction. Yes, it is pretty useful there.